oci/layer/utils.go

```diff
@@ -232,3 +227,14 @@ func InnerErrno(err error) error {
 	}
 	return errno
 }
+
+// isOverlayWhiteout returns true if the FileInfo represents an overlayfs style
+// whiteout (i.e. mknod c 0 0) and false otherwise.
+func isOverlayWhiteout(info os.FileInfo) bool {
+	sysStat := info.Sys().(*syscall.Stat_t)
```

Comment:
This would be cleaner if we did `unix.Lstat` and would remove the `syscall` dependency. I'm not sure if it'd be cleaner to pass the path to `isOverlayWhiteout` or a `unix.Stat_t` but we shouldn't be adding any new users of `syscall`.